projects/opensearch.org/package.yml

```diff
@@ -0,0 +1,49 @@
+distributable:
+  url: https://github.com/opensearch-project/OpenSearch/archive/refs/tags/{{version}}.tar.gz
+  strip-components: 1
+
+display-name: opensearch
+
+versions:
+  github: opensearch-project/OpenSearch
+
+dependencies:
+  openjdk.org: '*'
+
+build:
+  dependencies:
+    gradle.org: '*'
+  script:
+    - gradle -Dbuild.snapshot=false ":distribution:archives:no-jdk-{{hw.platform}}-tar:assemble"
+    - mkdir -p "{{prefix}}"
+    - tar --strip-components=1 -xf distribution/archives/no-jdk-{{hw.platform}}-tar/build/distributions/opensearch-*.tar.gz -C "{{prefix}}"
+    - run: |
+        sed -i.bak 's|#\s*cluster.name: .*|cluster.name: opensearch_pkgx|' "{{prefix}}/config/opensearch.yml"
+
+provides:
+  - bin/opensearch
+  - bin/opensearch-keystore
+  - bin/opensearch-plugin
+  - bin/opensearch-shard
+
+test:
+  dependencies:
+     gnu.org/coreutils: ^9
+     stedolan.github.io/jq: "*"
+     curl.se: "*"
+  script: |
+    opensearch-plugin list
```

Comment:
this might be simpler using arrays:
```yaml
    - opensearch-plugin list
    # While we'd love a good test like this, `initdb` doesn't run as root, and managing
    # all the environment passthrough with `sudo` is a nightmare.
    - run: opensearch -version |grep "Version: {{version}}"
      if: linux

    - run: |
        mkdir -p test/{data,logs}
        PORT=$(shuf -i 2000-65000 -n 1)
        opensearch -Ehttp.port=$PORT -Epath.data=$PWD/test/data -Epath.logs=$PWD/test/logs &
        pid=$!
        for i in $(seq 1 30); do
          curl -k --silent --fail http://localhost:$PORT/ > output.txt && break || sleep 1
        done
        kill $pid
        test "$(jq .version.number output.txt)" = \"{{version}}\"
      if: darwin
```